File: app/src/main/java/org/odddev/fantlab/auth/AuthRouter.java
package org.odddev.fantlab.auth;

import android.support.annotation.IdRes;

import org.odddev.fantlab.auth.login.LoginFragment;
import org.odddev.fantlab.auth.reg.RegFragment;
import org.odddev.fantlab.core.layers.router.Router;
import org.odddev.fantlab.core.utils.FragmentUtils;
import org.odddev.fantlab.home.HomeActivity;

/**
 * @author kenrube
 * @since 28.09.16
 */

public class AuthRouter extends Router<AuthActivity> {

    public AuthRouter(AuthActivity activity, @IdRes int containerId) {
        super(activity, containerId);
    }

    public void routeToLogin() {
        FragmentUtils.replaceFragment(getActivity(), getContainerId(), new LoginFragment(), true);
    }

    public void routeToReg() {
        FragmentUtils.replaceFragment(getActivity(), getContainerId(), new RegFragment(), true);
    }

    public void routeToHome(boolean loggedIn) {
        HomeActivity.start(getActivity(), loggedIn);
        getActivity().overridePendingTransition(0, 0);
    }
}

File: app/src/main/java/org/odddev/fantlab/auth/AuthRouter.kt
package org.odddev.fantlab.auth

import android.support.annotation.IdRes

import org.odddev.fantlab.auth.login.LoginFragment
import org.odddev.fantlab.auth.reg.RegFragment
import org.odddev.fantlab.core.layers.router.Router
import org.odddev.fantlab.core.utils.FragmentUtils
import org.odddev.fantlab.home.HomeActivity

/**
 * @author kenrube
 * *
 * @since 28.09.16
 */

class AuthRouter(activity: AuthActivity, @IdRes containerId: Int) : Router<AuthActivity>(activity, containerId) {

	fun routeToLogin() {
		FragmentUtils.replaceFragment(activity, containerId, LoginFragment(), true)
	}

	fun routeToReg() {
		FragmentUtils.replaceFragment(activity, containerId, RegFragment(), true)
	}

	fun routeToHome(loggedIn: Boolean) {
		HomeActivity.start(activity, loggedIn)
		activity.overridePendingTransition(0, 0)
	}
}
